src/db/commands.rs

```diff
@@ -66,19 +68,22 @@ pub fn fetch_messages(
         ..Default::default()
     };
 
+    let metrics = Rc::clone(metrics);
```

Comment:
I just noticed the rust team began recommending [Rc::clone vs .clone](https://github.com/rust-lang/rust/pull/42137#issuecomment-304908345) about a year ago, makes it clearer that this is a cheap Rc clone vs worrying about it being a different clone

Reply:
It makes sense for containers like RC (and I suppose vec may be similar), but I keep feeling that a given object should be responsible for making it's own clone, since it knows best how deep it needs to copy itself. I wonder if we'll see this pattern play out for other, similar structures?

Reply:
Not Vec but it makes sense for smart pointers. the original RFC (which suggested a new method new_ref instead that was rejected) mentioned Rc/Arc and the associated Weak pointer types. Here's a pretty good example it outlines:

https://github.com/nical/rfcs/blob/rc-newref-clone/text/0000-rc-newref-clone.md#example